summingbird-online/src/main/scala/com/twitter/summingbird/online/executor/AsyncBase.scala

```diff
@@ -34,6 +35,30 @@ object AsyncBase {
    * finished.
    */
   val OutstandingFuturesDequeueRatio = 2
+
+  /**
+   * Wait for n futures to finish. Doesn't block, the returned future is satisfied
+   * once n futures have finished either successfully or unsuccessfully.
+   * If n is greater than number of futures in queue then we wait on all of them.
+   */
+  def waitN[A](fs: Queue[Future[A]], n: Int): Future[Unit] = {
+    require(n >= 0)
```

Comment:
we could just return `Future.Unit` when `n <= 0` similar to how `drop(n)` is identity when `n <= 0`.